### PR TITLE
fix #2095 - new module verifybamid/verifybamid legacy 

### LINKED ISSUE #2095: new module: verifybamid/verifybamid
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by manifestoso**
## PR checklist

Closes #2095 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by jfy133** [comment]
Please check all TODOs are removed first :)

File: tests/modules/nf-core/verifybamid/verifybamid/nextflow.config [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// Including niche test data
+def test_niche_data_dir = "https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/delete_me"
+params {
+    niche_test {
+        'verifybamid' {
+            verifybamid_test_bam = "${test_niche_data_dir}/verifybamid/verifybam_bam.bam"
+        }
+    }
+}
```
Comment (Emiller88) [inline tests/modules/nf-core/verifybamid/verifybamid/nextflow.config:9]: Can we add a test for this?

**Review by Emiller88 — APPROVED** [review]
LGTM, but I'd love to see a test for the niche_test

File: modules/nf-core/verifybamid/verifybamid/main.nf [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    tuple val(meta), path("*.bestRG")                , optional:true, emit: bestrg
+    path "versions.yml"                                             , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def args_list = args.tokenize()
```
Comment (jfy133) [inline modules/nf-core/verifybamid/verifybamid/main.nf:30]: This splitting shouldn't be necessary, by defualt NXF will print them with spaces anyway (I assume left over from the earlier module). You can just directly put `$args` in the command.

File: modules/nf-core/verifybamid/verifybamid/main.nf [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    def bam_arg = "${bam}.endsWith('.bam|.BAM')" ? "--bam ${bam}" : ""
+    def bai_arg = "${bai}.endsWith('.bai|.BAI')" ? "--bai ${bai}" : ""
+    def refvcf_args = "${refvcf}.endsWith('.vcf|.vcf.gz')" ? "--vcf ${refvcf}" : ""
```
Comment (jfy133) [inline modules/nf-core/verifybamid/verifybamid/main.nf:34]: Can you run the tool without supply any input at all? I would assume bam  would at least be mandatory? In which case you should go directly into th ecommand.

File: modules/nf-core/verifybamid/verifybamid/meta.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  - qc
+  - contamination
+  - bam
+tools:
+  - "verifybamid":
+      description: "verifyBamID is a software that verifies whether the reads in particular file match previously known genotypes for an individual (or group of individuals), and checks whether the reads are contaminated as a mixture of two samples."
+      homepage: "https://genome.sph.umich.edu/wiki/VerifyBamID"
+      documentation: "http://genome.sph.umich.edu/wiki/VerifyBamID"
+      tool_dev_url: "https://github.com/statgen/verifyBamID"
+      doi: "https://doi.org/10.1016/j.ajhg.2012.09.004"
```
Comment (jfy133) [inline modules/nf-core/verifybamid/verifybamid/meta.yml:13]:
```suggestion
      doi: "10.1016/j.ajhg.2012.09.004"
```

File: modules/nf-core/verifybamid/verifybamid/main.nf (hunk 3 above)
Comment (jfy133) [reply modules/nf-core/verifybamid/verifybamid/main.nf:34]: Also given you don't specify bai/vcf as being optional in meta.yml I would assume all three are reuqired

File: tests/modules/nf-core/verifybamid/verifybamid/nextflow.config (hunk 1 above)
Comment (jfy133) [reply tests/modules/nf-core/verifybamid/verifybamid/nextflow.config:9]: yeah I'm not sure what's going on here. You don't need to make your own data_dir nor a list for a `delete_me` file - you just directly supply the rawlink to `file()` in the ftest itself.

**Review by jfy133 — CHANGES_REQUESTED** [review]

File: modules/nf-core/verifybamid/verifybamid/main.nf (hunk 2 above)
Comment (jfy133) [reply modules/nf-core/verifybamid/verifybamid/main.nf:30]: No? I don't believe so.

File: modules/nf-core/verifybamid/verifybamid/meta.yml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+      type: file
+      description: |
+        BAM file, a sorted, indexed, base quality recalibrated, and duplication-marked BAM file.
+        It also requires to contain "@RG" header lines to annotation different readGroups (sequencing runs and lanes).
+        The SM tag in the "@RG" header should match with one of the genotyped sample.
+      pattern: "*.bam"
+  - bai:
+      type: file
+      description: BAM index file BAI
+      pattern: "*.{bai}"
```
Comment (jfy133) [inline modules/nf-core/verifybamid/verifybamid/meta.yml:32]:
```suggestion
      pattern: "*.bai"
```

File: modules/nf-core/verifybamid/verifybamid/meta.yml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  - refvcf:
+      type: file
+      description: |
+        The input VCF file contains
```
Comment (jfy133) [inline modules/nf-core/verifybamid/verifybamid/meta.yml:36]: Is this optional?

File: tests/modules/nf-core/verifybamid/verifybamid/test.yml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+- name: verifybamid verifybamid test_verifybamid_verifybamid
+  command: nextflow run ./tests/modules/nf-core/verifybamid/verifybamid -entry test_verifybamid_verifybamid -c ./tests/config/nextflow.config -c ./tests/modules/nf-core/verifybamid/verifybamid/nextflow.config
+  tags:
+    - verifybamid
+    - verifybamid/verifybamid
+  files:
+    - path: output/verifybamid/test.log
```
Comment (jfy133) [inline tests/modules/nf-core/verifybamid/verifybamid/test.yml:7]: Any string inside it you can check for? e.g. 'run complete' or similar?

**Review by jfy133 — APPROVED** [review]
LGTM

**Comment by jfy133** [comment]
@manifestoso you can pull upstream from master and merge :+1:

**Pull request merged**